cmd/webhook/main.go

```diff
@@ -18,135 +18,122 @@ package main
-	"flag"
-	"fmt"
-	"log"
 
-	"go.uber.org/zap"
+	eventsv1alpha1 "github.com/google/knative-gcp/pkg/apis/events/v1alpha1"
+	messagingv1alpha1 "github.com/google/knative-gcp/pkg/apis/messaging/v1alpha1"
+	pubsubv1alpha1 "github.com/google/knative-gcp/pkg/apis/pubsub/v1alpha1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/tools/clientcmd"
+	"knative.dev/eventing/pkg/logconfig"
 	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection/sharedmain"
 	"knative.dev/pkg/logging"
-	"knative.dev/pkg/metrics"
 	"knative.dev/pkg/signals"
-	"knative.dev/pkg/system"
-	"knative.dev/pkg/version"
+	tracingconfig "knative.dev/pkg/tracing/config"
 	"knative.dev/pkg/webhook"
-
-	eventsv1alpha1 "github.com/google/knative-gcp/pkg/apis/events/v1alpha1"
-	messagingv1alpha1 "github.com/google/knative-gcp/pkg/apis/messaging/v1alpha1"
-	pubsubv1alpha1 "github.com/google/knative-gcp/pkg/apis/pubsub/v1alpha1"
+	"knative.dev/pkg/webhook/certificates"
+	"knative.dev/pkg/webhook/configmaps"
+	"knative.dev/pkg/webhook/resourcesemantics"
+	"knative.dev/pkg/webhook/resourcesemantics/defaulting"
+	"knative.dev/pkg/webhook/resourcesemantics/validation"
 )
 
-const (
-	component = "webhook"
-)
+var types = map[schema.GroupVersionKind]resourcesemantics.GenericCRD{
+	// For group messaging.cloud.google.com.
+	messagingv1alpha1.SchemeGroupVersion.WithKind("Channel"):   &messagingv1alpha1.Channel{},
+	messagingv1alpha1.SchemeGroupVersion.WithKind("Decorator"): &messagingv1alpha1.Decorator{},
 
-var (
-	masterURL  = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
-	kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
-)
+	// For group events.cloud.google.com.
+	eventsv1alpha1.SchemeGroupVersion.WithKind("Storage"):   &eventsv1alpha1.Storage{},
+	eventsv1alpha1.SchemeGroupVersion.WithKind("Scheduler"): &eventsv1alpha1.Scheduler{},
+	eventsv1alpha1.SchemeGroupVersion.WithKind("PubSub"):    &eventsv1alpha1.PubSub{},
 
-func SharedMain(resourceHandlers map[schema.GroupVersionKind]webhook.GenericCRD) {
+	// For group pubsub.cloud.google.com.
+	pubsubv1alpha1.SchemeGroupVersion.WithKind("PullSubscription"): &pubsubv1alpha1.PullSubscription{},
+	pubsubv1alpha1.SchemeGroupVersion.WithKind("Topic"):            &pubsubv1alpha1.Topic{},
+}
 
-	flag.Parse()
-	cm, err := configmap.Load("/etc/config-logging")
-	if err != nil {
-		log.Fatal("Error loading logging configuration:", err)
-	}
-	config, err := logging.NewConfigFromMap(cm)
-	if err != nil {
-		log.Fatal("Error parsing logging configuration:", err)
+func NewDefaultingAdmissionController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	// Decorate contexts with the current state of the config.
+	// store := defaultconfig.NewStore(logging.FromContext(ctx).Named("config-store"))
+	// store.WatchConfigs(cmw)
```

Comment:
Why are these lines commented? If they're unused but here for documentation, they should have a comment explaining why they exist and when they should be uncommented.

Reply:
I'm on your same page here. Don't know much about this webhook change... Copy/paste from [eventing](https://github.com/knative/eventing/blob/8e38791d3e4cba16df132c0b42ba68f62a746eac/cmd/webhook/main.go#L70). Will remove them.